R/template.R

```diff
@@ -1,110 +1,184 @@
+#' value of the `fn` or a single string of either `"first"` or `"second"`
+#' to pass along describing which level should be considered the `"event"`.
 #'
 #' @param case_weights For metrics supporting case weights, an unquoted
 #' column name corresponding to case weights can be passed here. If not `NULL`,
-#' the case weights will be passed on to `metric_fn` as the named argument
+#' the case weights will be passed on to `fn` as the named argument
 #' `case_weights`.
 #'
-#' @param ... Currently not used. Metric specific options are passed in
-#' through `metric_fn_options`.
 #'
-#' @param metric_fn_options A named list of metric specific options. These
+#' @param fn_options A named list of metric specific options. These
 #' are spliced into the metric function call using `!!!` from `rlang`. The
 #' default results in nothing being spliced into the call.
 #'
-#' @seealso [metric_vec_template()] [finalize_estimator()] [dots_to_estimate()]
-#'
+#' @keywords internal
 #' @export
-metric_summarizer <- function(metric_nm,
-                              metric_fn,
-                              data,
-                              truth,
-                              estimate,
-                              estimator = NULL,
-                              na_rm = TRUE,
-                              event_level = NULL,
-                              case_weights = NULL,
-                              ...,
-                              metric_fn_options = list()) {
+class_metric_summarizer <- function(name,
+                                    fn,
+                                    data,
+                                    truth,
+                                    estimate,
+                                    ...,
+                                    estimator = NULL,
+                                    na_rm = TRUE,
+                                    event_level = NULL,
+                                    case_weights = NULL,
+                                    fn_options = list()) {
+  rlang::check_dots_empty()
+
   truth <- enquo(truth)
   estimate <- enquo(estimate)
   case_weights <- enquo(case_weights)
 
-  validate_not_missing(truth, "truth")
-  validate_not_missing(estimate, "estimate")
+  # Explicit handling of length 1 character vectors as column names
+  nms <- colnames(data)
+  truth <- handle_chr_names(truth, nms)
+  estimate <- handle_chr_names(estimate, nms)
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = name,
+    .estimator = finalize_estimator(!! truth, estimator, name),
+    .estimate = fn(
+      truth = !! truth,
+      estimate = !! estimate,
+      !!! spliceable_estimator(estimator),
+      na_rm = na_rm,
+      !!! spliceable_event_level(event_level),
+      !!! spliceable_case_weights(case_weights),
+      !!! fn_options
+    )
+  )
+
+  dplyr::as_tibble(metric_tbl)
+}
+
+#' @rdname class_metric_summarizer
+#' @keywords internal
+#' @export
+numeric_metric_summarizer <- function(name,
```

Comment:
Slightest comment - I think it would make sense to put `numeric_metric_summarizer()` first in the file, so then the class/prob ones are grouped together (since they have the same argument structure and are generally more related)

Reply:
done in https://github.com/tidymodels/yardstick/pull/322/commits/08685c381095c148c6103554ea611c4e86fb89c9